### PR TITLE
Don't use std(in|out|err) as identifiers

### PR DESCRIPTION
**Pull request by ararslan**
Needed for https://github.com/JuliaLang/julia/pull/25959. In that PR, the lowercase names `stdin`, `stdout`, and `stderr` will be claimed by Base, so this avoids overwriting those names, which should hopefully allow documentation to build on that PR.

### CODE REVIEW

File: src/DocChecks.jl [hunk 1]
```diff
@@ -172,7 +172,7 @@ mutable struct Result
     file   :: String # File in which the doctest is written. Either `.md` or `.jl`.
     value  :: Any        # The value returned when evaluating `input`.
     hide   :: Bool       # Semi-colon suppressing the output?
-    stdout :: IOBuffer   # Redirected STDOUT/STDERR gets sent here.
+    io     :: IOBuffer   # Redirected STDOUT/STDERR gets sent here.
```
Comment (JeffBezanson) [inline src/DocChecks.jl]: It shouldn't be necessary to change field names.

Comment (ararslan) [reply src/DocChecks.jl]: Fair point, though arguably this is a better name anyway. ¯\\\_(ツ)\_/¯ I can change it back if that'd be preferable.

**Comment by fredrikekre** [comment]
The failure on nightly is fixed in https://github.com/JuliaDocs/Documenter.jl/pull/643/commits/31044e5d5015154069630741d3e597c75aebccbb

**Comment by ararslan** [comment]
Thanks for restarting CI, @mortenpi! Could you merge and tag?

**Pull request merged**